File: server/src/main/scala/producer/message_generator/deliver_after_generator/DeliverAfterGenerator.scala
package producer.message_generator.deliver_after_generator

import com.tools.teal.pulsar.ui.producer.v1.producer as pb
import producer.message_generator.data_generators.relative_date_time.RelativeDateTimeGenerator

case class DeliverAfterGenerator(
    generator: RelativeDateTimeGenerator
)

object DeliverAfterGenerator:
    def fromPb(v: pb.DeliverAfterGenerator): DeliverAfterGenerator =
        DeliverAfterGenerator(
            generator = RelativeDateTimeGenerator.fromPb(v.generator.get)
        )

    def toPb(v: DeliverAfterGenerator): pb.DeliverAfterGenerator =
        pb.DeliverAfterGenerator(
            generator = Some(RelativeDateTimeGenerator.toPb(v.generator))
        )

File: ui/components/ui/ProducerSession/producer-task/message-generator/deliver-after-generator/deliver-after-generator.ts
import * as pb from '../../../../../../grpc-web/tools/teal/pulsar/ui/producer/v1/producer_pb';
import { RelativeDateTimeGenerator, relativeDateTimeGeneratorFromPb, relativeDateTimeGeneratorToPb } from '../data-generators/relative-date-time/relative-date-time-generator';

export type DeliverAfterGenerator = {
  type: 'deliver-after-generator',
  generator: RelativeDateTimeGenerator
};

export function deliverAfterGeneratorFromPb(v: pb.DeliverAfterGenerator): DeliverAfterGenerator {
  return {
    type: 'deliver-after-generator',
    generator: relativeDateTimeGeneratorFromPb(v.getGenerator()!)
  };
}

export function deliverAfterGeneratorToPb(v: DeliverAfterGenerator): pb.DeliverAfterGenerator {
  const generatorPb = new pb.DeliverAfterGenerator();
  generatorPb.setGenerator(relativeDateTimeGeneratorToPb(v.generator));

  return generatorPb;
}
